hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -376,19 +379,44 @@ public Option<IndexedRecord> combineAndGetUpdateValue(IndexedRecord oldRecord, S
   }
 
   @Override
-  public Option<IndexedRecord> getInsertValue(Schema schemaIgnored, Properties propertiesIgnored) throws IOException {
+  public Option<IndexedRecord> getInsertValue(Schema schema, Properties propertiesIgnored) throws IOException {
     if (key == null || this.isDeletedRecord) {
       return Option.empty();
     }
 
-    HoodieMetadataRecord record = new HoodieMetadataRecord(key, type, filesystemMetadata, bloomFilterMetadata,
-        columnStatMetadata, recordIndexMetadata, secondaryIndexMetadata);
-    return Option.of(record);
+    if (schema == null || HoodieMetadataRecord.getClassSchema().equals(schema)) {
```

Comment:
if we are confident the schema has been interned, use `==` would be more efficient.

Reply:
if we can assume that the only possibilities for the schema for the special case of metadata payload are the `HoodieMetadataRecord`'s standard schema or that schema with meta-fields appended at the beginning, a simpler check can just be the number of fields being equal.

Reply:
> This will happen when first writing to the metadata table if you are looking for such a case.

then the schema should be just the global static schema of the class?

Reply:
`HoodieMetadataRecord.getClassSchema()` will return an instance of an object but parsing a string of the schema will not return the same exact instance. That was the reasoning for the special handling in the cache schema.